docker/docker-compose.yml

```diff
@@ -0,0 +1,142 @@
+# Auto generates the properties files for all services and the nginx config
+configs:
+  image: python:2.7
+  volumes:
+    - "./entrypoints:/entrypoints"
+    - "./configs/generators:/configs/generators"
+    - "$HOME/.aws/:/root/.aws:ro"
+    - "$HOME/.gu/grid/:/root/.gu/grid"
+    - "/configs"
+  entrypoint: /entrypoints/configs.sh
+
+# Proxy server for all services
+nginx:
+  image: nginx:latest
+  links:
+    - elasticsearch
+    - media-api
+    - thrall
+    - image-loader
+    - kahuna
+    - cropper
+    - metadata-editor
+    - usage
+    - collections
+    - imgops
+  ports:
+    - "80:80"
+    - "443:443"
+  volumes:
+    - "./entrypoints:/entrypoints"
+    - "./configs/nginx.conf:/etc/nginx/nginx.conf"
+  volumes_from:
+    - configs
+  entrypoint: /entrypoints/nginx.sh
+
+# Shared volume container
+data:
+  image: java:8
+  volumes:
+    - "../:/code"
+    - "/root/.ivy2"
```

Comment:
share the ivy cache with all containers so that they don't have to populate the cache themselves